umh-core/pkg/diagnosis/reduction_test.go

```
@@ -0,0 +1,234 @@
+const reduceHour = time.Hour
+
+// The six reductions and the three States. These specs drive a window, then
+// read the (v, State) pair back through Reduced.Get(); Reduced exposes no field
+// to read it from directly.
+var _ = Describe("Reduction", func() {
+	It("should reduce under the reduction it was given at construction, so two windows fed the same points under different reductions return different numbers", func() {
+		meanW, _ := NewWindow(reduceHour, 60*time.Second, Mean, false)
+		p95W, _ := NewWindow(reduceHour, 60*time.Second, P95, false)
+
+		t0 := time.Unix(1_000_000, 0)
+		for i := 0; i < 20; i++ {
```

golangci-lint (2.12.2)

[error] for loop can be changed to use an integer range (Go 1.22+)

(intrange)